simplify-basedomain-usage/README.md

```diff
@@ -0,0 +1,202 @@
+- For [the default apps](https://github.com/giantswarm/default-apps-aws) (apps deployed to both Management Clusters and Workload Clusters), we store the `baseDomain` in a special configmap called `$clustername-cluster-values` created by `cluster-apps-operator` in CAPI installations and by `cluster-operator` in `vintage`. This configmap [is referenced from the `App` CR of the default apps](https://github.com/giantswarm/default-apps-aws/blob/master/helm/default-apps-aws/templates/apps.yaml#L38-L40).
+- For other apps, we also store the `baseDomain` [in several app catalogs configuration](https://github.com/giantswarm/installations/blob/master/gohan/appcatalog/default-appcatalog-values.yaml#L5). That way the `baseDomain` is passed to all applications installed from those catalogs (Honeybadger has already stated that app catalog configuration may go away in the future).
+
+### Differences between `vintage` and CAPI installations
+
+The difference in the value for `baseDomain` between `vintage` and CAPI installations is sometimes confusing while working on applications that will be deployed to both kinds of installations.
+The fact that we reuse the same variable name for values that could potentially contain different things (does it contain info about the management cluster or not?) is confusing and error-prone, adding complexity where this value is used.
+
+### Different clusters in the same installation can't use different `baseDomain` values
+
+We have several controllers reconciling workload clusters, and these controllers receive the `baseDomain` configuration value to create resources for the reconciled clusters.
+Because we pass the `baseDomain` as a parameter to the controllers, all clusters in a single installation must share the same `baseDomain`. This has some big drawbacks
+
+- Different workload clusters managed by the same management cluster can't use different domains easily. If the customer wants a workload cluster with a different baseDomain, they would have to create a CNAME for the new domain and pass special configuration for that specific cluster (they could also create a new management cluster).
+- Workload clusters can't be moved to a different management cluster, the `baseDomain` creates a dependency between the management cluster and the workload cluster.
+
+### Same variable name for different values
+
+In our `installations` repository we save the `baseDomain` in the app catalog, so that this value is automatically passed to all apps from that catalog. For example, [on `goat` we have this value](https://github.com/giantswarm/installations/blob/master/gohan/appcatalog/default-appcatalog-values.yaml#L5)
+```yaml
+baseDomain: gtest.gigantic.io
+```
+
+But most applications expect the `baseDomain` to contain the cluster name as a prefix, like `mycluster.gtest.gigantic.io`.
+
+These differences here and there are confusing.
+
+## Proposal #1: Add `baseDomain` as an annotation to Cluster Custom Resource in CAPI installations
```

Comment:
Since the annotation would be set in `cluster-*` app (as that's where the Cluster CR is coming from), does this effectively mean that the source of truth for the base domain value is `cluster-*` app config?

Reply:
If here we mean `baseDomain` as the `dns-zone` containing the cluster name like `jose.gaws.gigantic.io`, then I'd say so, yeah.